spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+describe BooksController do
+  let!(:books) { create_list(:book, 3) }
+
+  describe 'GET #index' do
+    context 'With an unauthenticated user' do
+      it 'returns an unauthorized status' do
+        get :index, params: {}
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'With an authenticated user' do
+      include_context 'Authenticated User'
+
+      it 'without params returns all books' do
```

Comment:
You can add more contexts if you need to. If you are declaring an `it` using `with` or `without` as the first word, you are probably missing a context. In this case it could be
```ruby
context 'with an authenticated user' do
  context 'without params' do' do
    it 'returns all books' do
      ...
    end
  end
end
```